Add line graph to Provider Types report

PaulMorris opened the pull request:
Add a line graph to the Provider Types report, and add a color legend to all line graphs.  Currently that's Time to Review, Draft Applications, and Provider Types reports.

Tested by viewing the three reports, and confirming that the line graph reflects filtering on the provider types report.

![screenshot-2018-5-31 provider types](https://user-images.githubusercontent.com/1091693/40801918-2aed6876-64e2-11e8-8ed5-31c5b735f5b9.png)

Issue #784: Add Reporting based on provider types/taxonomies

PaulMorris commented:
The accessibility tests didn't like the white text on colored background in the legend.  This revision fixes that by moving the text outside the color swatches.  Also, I find that the legend is now less visually prominent with less risk of distracting from the graph itself.

![screenshot-2018-5-31 provider types 1](https://user-images.githubusercontent.com/1091693/40809481-a85252a2-64f8-11e8-9f62-051c56013104.png)

frankduncan commented on psm-app/cms-web/WebContent/WEB-INF/pages/admin/reports/provider_types.jsp:
```diff
@@ -62,13 +67,25 @@
                         <tr>
                           <th>Provider Type</th>
-                          <th class="providerTypesNum">Number Reviewed</th>
+                          <th class="providerTypesNum">Applications Reviewed</th>
```
Do we want to change the csv header as well?

PaulMorris replied:
Good call!  I made that change and added a commit to use the same wording in the "Time to Review" report for consistency.

frankduncan commented on psm-app/frontend/src/main/js/admin/providerTypesReport.js:
```diff
@@ -0,0 +1,108 @@
+    return a[1] < b[1];
+  });
+
+  // Get the ten types with the largest totals, and sort them alphabetically.
```
Do we want inline documentation of this?  It may be confusing to the user if they don't see everything they have selected without warning, but then it may be scope increasing too much to alert them with more than a generic "only the top ten results are displayed"

PaulMorris replied:
Good point!  I've added a message when there are more than 10 results.

![screenshot-2018-6-1 provider types](https://user-images.githubusercontent.com/1091693/40849364-c392e7e2-658f-11e8-9bb7-226cd4047167.png)

frankduncan commented on psm-app/cms-web/WebContent/WEB-INF/pages/admin/reports/provider_types.jsp:
```diff
@@ -50,6 +52,9 @@
             </div>
           </div>
+
+          <div id="providerTypesLineGraph"></div>
```
"Loading..." em again

PaulMorris replied:
Thanks, and done. :)

frankduncan commented on psm-app/frontend/src/main/js/admin/timeToReviewReport.js:
```diff
@@ -12,15 +12,18 @@ window.addEventListener("load", function drawTimeToReviewLineGraph() {
     "#timeToReviewLineGraph",
     "",
-    "Time to Review in Days (Mean in Blue and Median in Orange)",
+    "Time to Review in Days",
```
:+1:

PaulMorris commented:
Thanks for the review @frankduncan !  I've addressed all the comments.  Reviewed Documents report will get a legend when this lands, with no further changes needed.  :)
